### DISCUSSION TITLE
Split dataset on segment axis

### DISCUSSION THREAD

**Discussion by brsnw250** [etna-team/etna#327] [Improvements]
## Description

Add the possibility to split `TSDataset` on two disjoint sets of segments. Right now, the `TSDataset.train_test_split` method supports only temporal splits.

## Thougths
[Example](https://unit8co.github.io/darts/generated_api/darts.utils.model_selection.html) from darts

## Related

#328 
#310
